### DISCUSSION TITLE
How do you run a program at launch (-x|-e)

### DISCUSSION THREAD

**Discussion by ghost** [contour-terminal/contour#1660] [Q&A]
So I've been trying to figure out how to do something like launch contour to run something like contour -x tail -f /var/log/some.log. There is no man page and the help...doesn't heh. Running with -e it warns about only being a compatibility option for old KDE environments. That said it doesn't work and results in Aborted (core dumped). -x it doesn't understand at all. The help looks like you should just be able to toss the binary/script after it like contour coolbinfile but that results in "[error] Do not know what to do with `coolbinfile` will use it as a program" which works for some things but not tmux for example. Contour execute is depreciated so what am I missing to say run contour with a tmux session or just start a terminal with a script running in it?

**Comment by Yaraslaut** [comment]
Hi, you can pass executable directly like this `contour tmux` or set `tmux` as a shell for some profile, for example:
```
    tmux_shell:
        shell: "tmux"
```
and then run `contour profile tmux_shell`

**Reply by ghost** [reply]
I tried the syntax `contour pamix | tmux | etc` but it either would segfault or complain about not knowing what to do with the arg, but then say it will run it anyhow... Some things worked, somethings don't. So it's rather misleading that when it does work it complains. Anyhow thanks for the heads up.

**Reply by whisperity** [reply]
It's different levels of argument parsing. Usually to start a program inside Contour you need to say `contour execute foo` which will make the argument parsing logic put `foo` as the binary to spawn in the terminal. If you just say `contour foo` the complaint is that `foo` is not defined as a verb in the _argument parser_, and that the catch-all code is ran which defaults it to begin the binary (essentially placing `execute` in for you).

Otherwise, if you had a program named `set`, `contour set` would be ambiguous, as it could mean `contour` with `set` as a binary started OR `contour` frontend with the `set` action invoked in the argument parser.
The argument parser's things take priority, resolving the ambiguity. But that makes `set` as a binary name unusable directly, you need to explicitly go `contour execute set`.
